File: ios/Unlockly/Models/Wallet.swift
import Foundation

public struct Wallet: Codable {
    public var availableSeconds: Int
    public var earnedTodaySeconds: Int
    public var spentTodaySeconds: Int
    public var isProActive: Bool
    public var unfreezeExpirationTimestamp: Date?
    public var boostMultiplier: Float
    public var boostExpirationTimestamp: Date?
    public var emergencyUnlockUsedToday: Bool
    public var lastResetAt: Date

    public init(
        availableSeconds: Int = 0,
        earnedTodaySeconds: Int = 0,
        spentTodaySeconds: Int = 0,
        isProActive: Bool = false,
        unfreezeExpirationTimestamp: Date? = nil,
        boostMultiplier: Float = 1.0,
        boostExpirationTimestamp: Date? = nil,
        emergencyUnlockUsedToday: Bool = false,
        lastResetAt: Date = Date()
    ) {
        self.availableSeconds = availableSeconds
        self.earnedTodaySeconds = earnedTodaySeconds
        self.spentTodaySeconds = spentTodaySeconds
        self.isProActive = isProActive
        self.unfreezeExpirationTimestamp = unfreezeExpirationTimestamp
        self.boostMultiplier = boostMultiplier
        self.boostExpirationTimestamp = boostExpirationTimestamp
        self.emergencyUnlockUsedToday = emergencyUnlockUsedToday
        self.lastResetAt = lastResetAt
    }

    public var isUnfrozen: Bool {
        if let timestamp = unfreezeExpirationTimestamp {
            return Date() < timestamp
        }
        return false
    }

    public var isBoostActive: Bool {
        if boostMultiplier > 1.0, let timestamp = boostExpirationTimestamp {
            return Date() < timestamp
        }
        return false
    }

    public var effectiveBoostMultiplier: Float {
        isBoostActive ? boostMultiplier : 1.0
    }
}

File: app/src/main/java/com/antigravity/unlockly/data/model/Wallet.kt
package com.antigravity.unlockly.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "wallet")
data class Wallet(
    @PrimaryKey
    val id: Int = 1,
    val availableSeconds: Long = 0,
    val earnedTodaySeconds: Long = 0,
    val spentTodaySeconds: Long = 0,
    val productiveStudySecondsToday: Long = 0L,
    val isProActive: Boolean = false,
    val unfreezeExpirationTimestamp: Long = 0L,
    val boostMultiplier: Float = 1.0f,
    val boostExpirationTimestamp: Long = 0L,
    val emergencyUnlockUsedToday: Boolean = false,
    val lastResetAt: Long = System.currentTimeMillis()
) {
    val isUnfrozen: Boolean
        get() = System.currentTimeMillis() < unfreezeExpirationTimestamp

    val isBoostActive: Boolean
        get() = boostMultiplier > 1.0f && System.currentTimeMillis() < boostExpirationTimestamp

    val effectiveBoostMultiplier: Float
        get() = if (isBoostActive) boostMultiplier else 1.0f
}
